libraries/AP_HAL_SITL/SITL_State.cpp

```diff
@@ -212,6 +212,18 @@ void SITL_State::wait_clock(uint64_t wait_time_usec)
             usleep(1000);
         }
     }
+    // check the outbound TCP queue size.  If it is too long then
+    // MAVProxy/pymavlink take too long to process packets and it ends
+    // up seeing traffic well into our past and hits time-out
+    // conditions.
+    while (true) {
+        const int queue_length = hal.uartA->get_system_outqueue_length();
+        // ::fprintf(stderr, "queue_length=%d\n", (signed)queue_length);
+        if (queue_length < 1024) {
+            break;
+        }
+        usleep(1000);
```

Comment:
I'd like this to only take effect is speedup>1

Reply:
Done.